### PR TITLE
CLAS: Avoid unnecessary serialize of testclasses include

### PR DESCRIPTION
**Pull request by mbtools**
Unit tests are only functional if the corresponding flag is set (`seoclassdf-with_unit_tests`). Therefore, if the flag is not set, there's no need to serialize this include.

### CODE REVIEW

**Comment by larshp** [comment]
`seoclassdf` is buffered, but I think it should already be in memory somewhere in the call stack?

File: src/objects/zcl_abapgit_oo_serializer.clas.abap [hunk 1]
```diff
@@ -297,6 +297,15 @@ CLASS ZCL_ABAPGIT_OO_SERIALIZER IMPLEMENTATION.
 
   METHOD serialize_testclasses.
 
+    DATA lv_with_unit_test TYPE abap_bool.
+
+    " Unit tests are only functional if the corresponding flag is set
+    " Therefore, if the flag is not set, there's no need to serialize this include
+    SELECT SINGLE with_unit_tests FROM seoclassdf INTO lv_with_unit_test WHERE clsname = is_clskey-clsname.
```
Comment (flaiker) [inline src/objects/zcl_abapgit_oo_serializer.clas.abap:304]: key field version is not supplied?

Comment (mbtools) [reply src/objects/zcl_abapgit_oo_serializer.clas.abap]: done

**Comment by mbtools** [comment]
The class properties are selected only after the coding. I changed to the same function call which provides its own buffer

**Comment by larshp** [comment]
thanks

It looks like `mv_skip_testclass` should be set to true? Plus after spending 5 minutes reading the code for `SEO_CLIF_GET`, its not totally obvious to me that it does caching, so just keep the SELECT, its easier to understand?

**Comment by mbtools** [comment]
The function is much faster on the second run so it certainly looks like it buffers. I set the skip flag although the result is the same
https://github.com/abapGit/abapGit/blob/master/src/objects/zcl_abapgit_object_clas.clas.abap#L705

**Comment by larshp** [comment]
looking at the code its not obvious that SEO_CLIF_GET is called twice, and also not obvious it does caching 🤷‍♂️

**Pull request merged**